File: leetcode/counting-bits.py
from typing import List


class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0] * (n + 1)

        for i in range(1, n + 1):
            # Brian Kernighan's algorithm
            res[i] = 1 + res[i & i - 1]

        return res

File: leetcode/counting-bits.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
 public:
    vector<int> countBits(int n)
    {
        vector<int> res(n + 1);

        for (auto i = 1; i <= n; ++i) {
            // Brian Kernighan’s Algorithm
            res[i] = 1 + res[i & (i - 1)];
        }

        return res;
    }
};
